components/js/slickGrid.ts

```diff
@@ -500,6 +501,9 @@ export class SlickGrid implements OnChanges, OnInit, OnDestroy, AfterViewInit {
         this._grid.onContextMenu.subscribe((e, args) => {
             this.onContextMenu.emit(e);
         });
+        this._grid.onBeforeAppendCell.subscribe((e, args) => {
+            return this.getAdditionalCssClassesForCell ? this.getAdditionalCssClassesForCell(args.row, args.cell) : undefined;
```

Comment:
Whats the point of this? This seems _REALLY_ specific use case. Wouldn't it just be better to expose an event and let the caller handle exactly what they are trying to do?